webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +122,16 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                document.location.href = next;
```

Comment:
One thing we want to do is make sure we're redirecting to a url that's explicitly on this server and not a url on some other server. My JS is lacking, so I asked @Osmose and he suggested something like this:

```
document.location.href = URL(next).pathname;
```

If you need to pull in the query params, those are in `.searchParams`.

Further, he pointed out you can pull out the next param from this url's query string with this:

```
next = URL(window.location.href).searchParams.next;
```

If that works (I haven't tested it), then you can ditch the function you pulled from StackOverflow.